core/station/impl/src/services/account.rs

```diff
@@ -248,23 +265,41 @@ impl AccountService {
             // if this is the first account created, and there is no cycle minting account set, set this account as the cycle minting account
             if system_info.get_cycle_obtain_strategy() == &CycleObtainStrategy::Disabled
                 && ACCOUNT_REPOSITORY.len() == 1
-                && matches!(new_account.blockchain, Blockchain::InternetComputer)
-                && new_account.standard == BlockchainStandard::Native
-                && new_account.symbol == "ICP"
             {
-                ic_cdk::println!("Setting cycle minting account to {}", uuid);
-
-                system_info.set_cycle_obtain_strategy(CycleObtainStrategy::MintFromNativeToken {
-                    account_id: *uuid.as_bytes(),
-                });
-                write_system_info(system_info);
-
-                #[cfg(target_arch = "wasm32")]
-                crate::services::SYSTEM_SERVICE.set_fund_manager_obtain_cycles(
-                    &CycleObtainStrategy::MintFromNativeToken {
-                        account_id: new_account.id,
-                    },
-                );
+                // find the mainnet ICP asset that minting can be done from
+                if let Some(icp_asset) = self.asset_repository.list().iter().find(|asset| {
+                    asset.blockchain == Blockchain::InternetComputer
+                        && asset
+                            .standards
+                            .contains(&TokenStandard::InternetComputerNative)
+                        && asset.metadata.contains(&MetadataItem {
+                            key: TokenStandard::METADATA_KEY_LEDGER_CANISTER_ID.to_string(),
+                            value: MAINNET_LEDGER_CANISTER_ID.to_string(),
+                        })
+                }) {
+                    // check if the new account has the ICP asset
+                    if new_account
+                        .assets
+                        .iter()
+                        .any(|account_asset| account_asset.asset_id == icp_asset.id)
+                    {
+                        ic_cdk::println!("Setting cycle minting account to {}", uuid);
+
+                        system_info.set_cycle_obtain_strategy(
+                            CycleObtainStrategy::MintFromNativeToken {
+                                account_id: *uuid.as_bytes(),
```

Comment:
shouldn't this use the seed?

Reply:
No, because this has to store which account will be used to top up the station. On the other hand ObtainCyclesOptions stores the SubAccount, which was indeed created wrongly using the account_id. Good catch, fixed!